mne_features/univariate.py

```diff
@@ -975,6 +978,51 @@ def compute_svd_entropy(data, tau=2, emb=10):
     return -np.sum(np.multiply(sv_norm, np.log2(sv_norm)), axis=-1)
 
 
+
+def compute_powercurve_deviation(data, sfreq, freq_range=[0.1, 50.]):
+    """ Linear fit to the log-log frequency-curve with fit error (per channel).
+    Parameters
+    ----------
+    data : ndarray, shape (n_channels, n_times)
+
+    freq_range: list, length 2
+        Gives floats fmin, fmax that define frequency range of the linear fit.
+
+    sfreq: float
+        The sampling frequency.
+    Returns
+    -------
+    output : ndarray, shape (n_channels, 4)
+        The 4 columns are slope, intercept, mse, and R2 per channel.
+    """
+    ps, freqs = power_spectrum(sfreq, data, return_db=False)
+
+    # mask limiting to input freq_range
+    mask = np.logical_and(freqs >= freq_range[0], freqs <= freq_range[1])
+
+    # freqs and ps selected over input freq_range and expressed in log scale
+    freqs, ps = 10 * np.log10(freqs[mask]), 10 * np.log10(ps[:, mask])
+
+    # linear fit
+    lm = LinearRegression()
+    fit_info = []
```

Comment:
You already know the shape of the array the function will return. Therefore, instead of using a list, you could pre-allocate with `fit_info = np.empty((n_channels, 4))` and fill the array with: 
```
for idx, power in enumerate(ps):
    ...
    fit_info[idx, :] = np.array([coef, intercept, mse, r2])
return fit_info.ravel()
```